src/libraries/System.Data.Common/src/System/Xml/XPathNodePointer.cs

```diff
@@ -67,9 +64,10 @@ private XPathNodeType DecideXPNodeTypeForTextNodes(XmlNode node)
             //the function can only be called on text like nodes.
             Debug.Assert(XmlDataDocument.IsTextNode(node.NodeType));
             XPathNodeType xnt = XPathNodeType.Whitespace;
-            while (node != null)
+            XmlNode? n = null;
```

Comment:
Doesn't this need to be `XmlNode? n = node;`? As written, I don't think you will ever get into the loop.

Reply:
Damn. Good catch! Fixed.